libnavui-speedlimit/src/main/java/com/mapbox/navigation/ui/speedlimit/api/MapboxSpeedInfoApi.kt

```diff
@@ -0,0 +1,55 @@
+package com.mapbox.navigation.ui.speedlimit.api
+
+import com.mapbox.navigation.base.formatter.DistanceFormatterOptions
+import com.mapbox.navigation.core.trip.session.LocationMatcherResult
+import com.mapbox.navigation.ui.base.formatter.ValueFormatter
+import com.mapbox.navigation.ui.speedlimit.SpeedLimitAction
+import com.mapbox.navigation.ui.speedlimit.SpeedLimitProcessor
+import com.mapbox.navigation.ui.speedlimit.SpeedLimitResult
+import com.mapbox.navigation.ui.speedlimit.model.PostedAndCurrentSpeedFormatter
+import com.mapbox.navigation.ui.speedlimit.model.SpeedData
+import com.mapbox.navigation.ui.speedlimit.model.SpeedInfoValue
+
+/**
+ * Mapbox implementation of for deriving the posted speed limit and current speed at users current
+ * location.
+ *
+ * @param processor an instance of a [SpeedLimitProcessor]
+ */
+class MapboxSpeedInfoApi internal constructor(private val processor: SpeedLimitProcessor) {
+
+    /**
+     * Mapbox implementation of for deriving the posted speed limit and current speed at users
```

Comment:
Do we need both "of" and "for" here?
Also, maybe "user's current location" or just "user current location"?